editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/properties/bean/PropertyMethodValidator.java

```diff
@@ -0,0 +1,68 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Red Hat, Inc. 
+ * Distributed under license by Red Hat, Inc. All rights reserved. 
+ * This program is made available under the terms of the 
+ * Eclipse Public License v1.0 which accompanies this distribution, 
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/
+package org.fusesource.ide.camel.editor.properties.bean;
+
+import org.eclipse.core.databinding.observable.map.IObservableMap;
+import org.eclipse.core.databinding.validation.IValidator;
+import org.eclipse.core.databinding.validation.ValidationStatus;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.IType;
+import org.eclipse.jdt.core.JavaModelException;
+import org.fusesource.ide.camel.editor.globalconfiguration.beans.BeanConfigUtil;
+import org.fusesource.ide.camel.editor.internal.CamelEditorUIActivator;
+import org.fusesource.ide.camel.editor.utils.CamelUtils;
+import org.fusesource.ide.camel.model.service.core.model.CamelBean;
+import org.fusesource.ide.foundation.core.util.Strings;
+
+/**
+ * @author brianf
+ *
+ */
+public class PropertyMethodValidator implements IValidator {
+	
+	private IObservableMap<?, ?> modelMap;
+	private BeanConfigUtil beanConfigUtil = new BeanConfigUtil();
+	
+	public PropertyMethodValidator(IObservableMap<?, ?> modelMap) {
+		this.modelMap = modelMap;
+	}
+	
+	@Override
+	public IStatus validate(Object value) {
+		Object control = modelMap.get(CamelBean.PROP_CLASS);
+		if (control != null) {
+			String className = (String) control;
+			String methodName = (String) value;
+			IJavaProject jproject = beanConfigUtil.getJavaProject(CamelUtils.project());
```

Comment:
better to avoid to use the CamelUtils.project() as it is linking this code the workbench state.
On AbstractCamelElement we can getCamelFile().getResource().getProject()